tests/core/TransactionSelector.spec.ts

```diff
@@ -205,6 +205,28 @@ describe('TransactionSelector', () => {
       ];
       expect(result).toEqual(expected);
     });
+
```

Comment:
Shouldn't we have new tests for the changes in the functions ````getNumberOfOperationsAndTransactionsAlreadyInBlock ```` and ````getHighestFeeTransactionsFromCurrentTransactionTime ```` and not this one? The following test will exercise the error in the first function but not the 2nd one right?

Reply:
It will go into both errors because the transactions passed in as argument will trigger one, and the transactions read from the DB will trigger another. I didn't add tests for the individual functions because they are private and I am able to cover the branches by testing the public function so I did it that way. I can add tests for the private functions too though.

Reply:
Personally I like testing the smaller functions as it is easier to add variations to the test. If there's a change in the private function, we will have to change the inputs here to match the both the new variation and the other private function as well.

Also since our unit tests are white box, it is ok to test the private functions separately.

But you have already written the tests to test all branches I am ok with this change.